EIPS/eip-7093.md

```diff
@@ -0,0 +1,409 @@
+- **RecoveryAccount**: This component encapsulates the core of the social recovery functionality. It is designed to be flexible, composable, and extensible to adapt to various recovery needs. Each RecoveryAccount is defined by an instance contract, crafted by smart contract developers, which embeds the essential logic for the recovery process.
+
+![social_recovery_flow](../assets/eip-7093/social-recovery-flow.png)
+
+### DataTypes
+
+### `TypesAndDecoders.sol`
+
+This defines the necessary data types required by this interface standard.
+
+```solidity
+pragma solidity ^0.8.13;
+// SPDX-License-Identifier: MIT
+
+/**
+ * @dev Structure representing an identity with its signature/proof verification logic
+ * Represents an EOA/CA account when signer is empty,use : @openzeppelin/contracts
+ * /utils/cryptography/SignatureChecker.isValidSignatureNow(guardianVerifier,hash,signature) for signature verification.
+ * OtherWise execute IGuardianPermissionVerifier(guardianVerifier).isValidPermissions(signer,hash,signature).
+ */
+struct Identity {
+    address guardianVerifier;
+    bytes signer;
+}
+
+/**
+ * @dev Structure representing a guardian with a property
+ * The property of Guardian are defined by the associated RecoveryPolicyVerifier contract.
+ */
+struct GuardianInfo {
+    Identity guardian;
+    uint32 property; //eg.,Weight,Percentage,Role with weight,etc.
+}
+
+/**
+ * @dev Structure representing a threshold configuration
+ */
+struct ThresholdConfig {
+    uint32 threshold; // Threshold value
+    int48 lockPeriod; // Lock period for the threshold
+}
+
+/**
+ * @dev Structure representing a recovery configuration
+ * A RecoveryConfig can have multiple threshold configurations for different threshold values and their lock periods
+ */
+struct RecoveryConfig {
+    GuardianInfo[] guardianInfos;
+    ThresholdConfig[] thresholdConfigs;
+}
+
+/**
+ * @dev Structure representing a recovery policy with its verification logic (policy verifier contract address)
+ */
+struct RecoveryPolicy {
+    address policyVerifier;
+    RecoveryConfig config;
+}
+
+struct Permission {
+    Identity guardian;
+    bytes signature;
+}
+
+```
+
+The `Identity` structure represents various types of guardians. The process of identity verification is as follows:
+
+- When the `signer` value in the declared entity is empty, this implies that the `Identity` entity is of EOA/SCA account type. In this case, `guardianVerifier` address should be the address of EOA/SCA (the actual signer). For permission verification of this `Identity` entity, one can use `SignatureChecker.isValidSignatureNow` from the Openzeppelin library, which supports both ECDSA and [ERC-1271](./eip-1271.md) signature verifications.
```

Comment:
Don't make recommendations on implementation. Instead describe the properties you'd like the implementation to have. For example, you may want to recommend implementers watch out for signature malleability attacks (which is what is provided by OZ as far as I know.)